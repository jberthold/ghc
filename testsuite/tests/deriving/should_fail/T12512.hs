{-# LANGUAGE DataKinds #-}
{-# LANGUAGE KindSignatures #-}
{-# LANGUAGE StandaloneDeriving #-}
{-# LANGUAGE UnboxedTuples #-}
module T12512 where

import GHC.Exts

class Wat1 (a :: TYPE 'UnboxedTupleRep)
deriving instance Wat1 (# a, b #)
