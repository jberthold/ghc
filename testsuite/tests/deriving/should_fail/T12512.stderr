
T12512.hs:10:1: error:
    • Can't make a derived instance of ‘Wat1 (# a, b #)’:
        The last argument of the instance cannot be an unboxed tuple
    • In the stand-alone deriving instance for ‘Wat1 (# a, b #)’
